### PR TITLE
fix(dev-infra): correctly check for commit that closes PR

### PR DESCRIPTION
**Pull request by gkalpak**
The `ng-dev release publish` command needs to check whether a commit closed a pull request. This is implemented via checking the commit message for specific closing keywords referencing the pull request number.

The regex used previously failed to correctly ensure that the specified pull request was referenced. For example, it would allow `#12345` to also match for `#1234`.

This PR fixes the regex.

##
This PR also includes a `refactor` commit that addresses my feedback from https://github.com/angular/angular/pull/38656#pullrequestreview-482129333.

### CODE REVIEW

File: dev-infra/release/publish/external-commands.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export async function invokeSetNpmDistCommand(npmDistTag: string, version: semve
     info(green(`  ✓   Set "${npmDistTag}" NPM dist tag for all packages to v${version}.`));
   } catch (e) {
     error(e);
-    error(red(`  ✘   An error occurred while setting the NPM dist tag for ${npmDistTag}.`));
+    error(red('  ✘   An error occurred while setting the NPM dist tag for "${npmDistTag}".'));
```
Comment (josephperrott) [inline dev-infra/release/publish/external-commands.ts:47]: This needs to remain a template string to allow for `${npmDistTag}` correct?

File: dev-infra/release/publish/pull-request-state.ts [hunk 2]
```diff
@@ -68,5 +68,5 @@ async function isCommitClosingPullRequest(api: GitClient, sha: string, id: numbe
   const {data} = await api.github.repos.getCommit({...api.remoteParams, ref: sha});
   // Matches the closing keyword supported in commit messages. See:
   // https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords.
-  return data.commit.message.match(new RegExp(`close[sd]? #${id}[^0-9]?`, 'i'));
+  return data.commit.message.match(new RegExp(`close[sd]? #${id}(?!\\d)`, 'i'));
```
Comment (AndrewKushnir) [inline dev-infra/release/publish/pull-request-state.ts:71]: It looks like we cover `close`, `closes` and `closed` keywords here. Should we also add support for other ones? If not, can we have a comment here explaining why that's the case?

**Review by AndrewKushnir — APPROVED** [review]
LGTM (assuming Joy's comment is addressed) 👍

**Comment by gkalpak** [comment]
For reference, I created #39229 to address the review feedback.

**Pull request closed without merging**